gateway-service-definitions/src/main/resources/services/yarnuiv2/3.0.0/rewrite.xml

```diff
@@ -171,12 +171,34 @@
     </content>
   </filter>
    
-  <rule dir="IN" name="YARNUIV2/yarnuiv2/inbound/proxy" pattern="*://*:*/**/yarnuiv2/proxy/{**}?{**}">
-      <rewrite template="{$serviceUrl[YARNUIV2]}/proxy/{**}?{**}"/>
-   </rule>
+  <filter name="YARNUIV2/yarnuiv2/outbound/headers/jobhistory/job">
+    <content type="application/x-http-headers">
+      <apply path="Location" rule="YARNUIV2/yarnuiv2/outbound/headers/jobhistory/job/location"/>
+    </content>
+  </filter>
+
+  <rule dir="OUT" name="YARNUIV2/yarnuiv2/outbound/proxy" pattern="*://*:*/proxy/{**}">
+    <rewrite template="{$frontend[url]}/yarnuiv2/proxy/{**}"/>
```

Comment:
Should be using {$serviceUrl[YARNUIV2]} instead of {$frontend[url]}

Reply:
I see references to frontendurl in the existing rewrite file itself? Any specific reason that, this needs to be changed?

Reply:
{$serviceUrl[YARNUIV2]} is the preferred way. You want to include the specific URL that was supplied in the topology file, that way changing the URL in topology will update URL's all the rules.

Reply:
@moresandeep : I tried changing {$frontend[url]} with {$serviceUrl[YARNUIV2]} and I am not sure why, but the fix stopped working. So I am leaving it with frontendurl itself.

Reply:
Interesting, is it because  {$serviceUrl[YARNUIV2]} adds the absolute url ? some more information on the failure would be good to have.
Thanks.